### PR TITLE
Adds a step to start VMs during admin upgrade

### LINKED ISSUE #687: Remediate deallocated VMs
We have a cluster where the subscription was suspended and the VMs automatically deallocated.  Now the subscription is good again.

We need to detect this and power the VMs on.  Another step like the pull secret fix on upgrade.

### PR DESCRIPTION
**Pull request by m1kola**
ref #687 (partial fix)

### CODE REVIEW

File: pkg/install/start_vms.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	// TODO: Check if customer can create VMs in the cluster resource group (deny assignment?)
+	vms, err := i.virtualmachines.List(ctx, resourceGroupName)
+	if err != nil {
+		return err
+	}
+
+	deallocatedVMs := make([]mgmtcompute.VirtualMachine, 0, len(vms))
+	for _, vm := range vms {
+		// TODO: consider getting instance view concurrently with errgroup
+		instanceView, err := i.virtualmachines.InstanceView(ctx, resourceGroupName, *vm.Name)
```
Comment (jim-minter) [inline pkg/install/start_vms.go:26]: fwiw pkg/frontend/admin_openshiftcluster_resources_list.go does vmClient.Get(ctx, clusterResourceGroup, *res.Name, mgmtcompute.InstanceView) to much the same effect

File: pkg/install/start_vms.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+			if strings.HasPrefix(*status.Code, "PowerState") {
+				if *status.Code == "PowerState/deallocated" {
+					deallocatedVMs = append(deallocatedVMs, vm)
+				}
+				break
+			}
+		}
+	}
+
+	// TODO: consider starting VMs concurrently with errgroup
```
Comment (jim-minter) [inline pkg/install/start_vms.go:46]: definitely do that

**Comment by m1kola** [comment]
Rebased on top of master (d632f8088ffc2e565bfb63e2b1282a53c7ff76d4).

File: pkg/install/start_vms.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+			continue
+		}
+
+		for _, status := range *vm.VirtualMachineProperties.InstanceView.Statuses {
+			if status.Code == nil {
+				continue
+			}
+
+			// Ref: https://docs.microsoft.com/en-us/azure/virtual-machines/windows/states-lifecycle
+			if strings.HasPrefix(*status.Code, "PowerState") {
```
Comment (jim-minter) [inline pkg/install/start_vms.go:55]: seems a little surplus to requirements

File: pkg/install/start_vms.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+func (i *Installer) startVMs(ctx context.Context) error {
+	resourceGroupName := stringutils.LastTokenByte(i.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+	vms, err := i.virtualmachines.List(ctx, resourceGroupName)
+	if err != nil {
+		return err
+	}
+
+	{
+		g, groupCtx := errgroup.WithContext(ctx)
+		for idx, vm := range vms {
```
Comment (jim-minter) [inline pkg/install/start_vms.go:26]: as discussed, let's go with this for now.  I still worry about the case of lots of parallel admin upgrades * lots of VMs * getting 409s from Azure.

**Pull request merged**